File: src/jk_5/nailed/map/gameloop/instructions/InstructionSetWinner.java
package jk_5.nailed.map.gameloop.instructions;

import jk_5.nailed.map.gameloop.GameThread;
import jk_5.nailed.map.gameloop.IInstruction;

/**
 * No description given
 *
 * @author jk-5
 */
public class InstructionSetWinner implements IInstruction {

    private String winner;

    @Override
    public void injectArguments(String arguments) {
        this.winner = arguments;
    }

    @Override
    public void execute(GameThread controller) throws InterruptedException {
        controller.setWinner(controller.getMap().getTeamManager().getTeam(this.winner));
    }
}

File: src/jk_5/nailed/map/gameloop/instructions/InstructionSetWinner.scala
package jk_5.nailed.map.gameloop.instructions

import jk_5.nailed.map.gameloop.GameThread
import jk_5.nailed.map.gameloop.IInstruction

/**
 * No description given
 *
 * @author jk-5
 */
class InstructionSetWinner extends IInstruction {
  private var winner: String = null
  def injectArguments(arguments: String) = this.winner = arguments
  def execute(controller: GameThread) = controller.getMap.getTeamManager.getTeam(this.winner).foreach(t=>controller.setWinner(t))
}
